Setup CI via GitHub Actions

simonkassel opened the pull request.

simonkassel committed:
test github action

CloudNiner committed:
Rename curblr dir to pycurb

CloudNiner committed:
Rename CurbLRObject to PyCurbObject

CloudNiner committed:
Add flake8 config and lint

CloudNiner committed:
Format with yapf

CloudNiner committed:
Add stub test so there's something to run

CloudNiner committed:
Add codecov

CloudNiner commented:
Moved `curblr` dir to `pycurb`, applied flake8 and yapf formatting to all python files, added a stub test case and re-enabled codecov.

You should still be able to load up a curblr json file in pycurb, to do so, cd to this repo root dir wherever you have it checked out to and in a python or ipython shell try:
```
import pycurb
curblr = pycurb.FeatureCollection.from_file("./path/to/curblr.json")
curblr.features[0].to_dict()
```

CloudNiner commented:
Closes #7

CloudNiner committed:
Expand README with some basic documentation

The documentation added here highlights the
translation layer between python objects
and JSON. Most of the other features currently
available are better documented via docstrings.

simonkassel commented:
@CloudNiner Finally getting to this. thanks, it's looking good to me. Renaming to curblr to pycurb totally makes sense. I had one issue/improvement re: running the tests locally which I addressed in more detail here https://github.com/azavea/PyCurb/pull/13

simonkassel committed:
Merge pull request #12 from azavea/feature/awf/docs

Expand README with some basic documentation

simonkassel committed:
fix bug in fill_unregulated_time method

simonkassel committed:
add venv comment to README
